src/runtime/arc.ts

```diff
@@ -591,7 +588,16 @@ ${this.activeRecipe.toString()}`;
       assert(storageKey, `couldn't find storage key for handle '${recipeHandle}'`);
       const type = recipeHandle.type.resolvedType();
       assert(type.isResolved());
-      const store = await this.storageProviderFactory.connect(recipeHandle.id, type, storageKey);
+      let store = await this.storageProviderFactory.connect(recipeHandle.id, type, storageKey);
+
+      if (!store && recipeHandle.fate === 'map') {
+        assert(recipeHandle.fate === 'map');
+        const storeRef = this.findStoreById(recipeHandle.id);
+        assert(storeRef instanceof StorageStub);
+        if (storeRef instanceof StorageStub) {
+          store = await storeRef.inflate();
+        }
```

Comment:
the asserts here are redundant with the if-checks; have one or the other but not both.

Reply:
these were leftovers, removed, thanks!